### ISSUE TITLE
64 bit integer arithmetic?

### ISSUE THREAD

**Issue by chriselrod** [google/highway#907]
For example, this works:
https://gcc.godbolt.org/z/Kn8jce6nb
but replacing `using T = int32_t` with `using T = int64_t` results in compilation errors.

I know that many architectures don't support it (e.g., only AVX3/AVX512 supports 64 bit integer multiplication), but it can be replicated with high/low sequences of 32 bit operations.

Would be nice to have this work automatically.

**Comment by jan-wassenberg** [comment]
Hi @chriselrod, the good news is that SVE and RVV also support i64 * i64, so an argument can be made that previous ISAs should emulate the modern ones.

Would you be willing to send a patch for this? Here's one example of adding an op: https://github.com/google/highway/commit/18f98b9133e1c25633282f949d218fffec08b3aa
One could also use the approach from https://github.com/google/highway/commit/7d080f1dcdb798dd8661951aca3aa9b0a5dd352c to implement the fallback only once using generic ops (MulEven?), if that makes sense.
